client/keystore/src/local.rs

```diff
@@ -140,19 +83,49 @@ impl Keystore for LocalKeystore {
 	/// If the `[seed]` is `Some` then the key will be ephemeral and stored in memory.
 	fn sr25519_generate_new(
 		&self,
-		id: KeyTypeId,
+		key_type: KeyTypeId,
 		seed: Option<&str>,
 	) -> std::result::Result<sr25519::Public, TraitError> {
 		let pair = match seed {
-			Some(seed) =>
-				self.0.write().insert_ephemeral_from_seed_by_type::<sr25519::Pair>(seed, id),
-			None => self.0.write().generate_by_type::<sr25519::Pair>(id),
+			Some(seed) => self
+				.0
+				.write()
+				.insert_ephemeral_from_seed_by_type::<sr25519::Pair>(seed, key_type),
+			None => self.0.write().generate_by_type::<sr25519::Pair>(key_type),
 		}
 		.map_err(|e| -> TraitError { e.into() })?;
 
 		Ok(pair.public())
 	}
 
+	fn sr25519_sign(
+		&self,
+		key_type: KeyTypeId,
+		public: &sr25519::Public,
+		msg: &[u8],
+	) -> std::result::Result<Option<sr25519::Signature>, TraitError> {
+		let res = self
+			.0
+			.read()
+			.key_pair_by_type::<sr25519::Pair>(public, key_type)?
+			.map(|pair| pair.sign(msg));
+		Ok(res)
+	}
+
+	fn sr25519_vrf_sign(
+		&self,
+		key_type: KeyTypeId,
+		public: &Sr25519Public,
+		transcript_data: VRFTranscriptData,
+	) -> std::result::Result<Option<VRFSignature>, TraitError> {
+		let res = self.0.read().key_pair_by_type::<Sr25519Pair>(public, key_type)?.map(|pair| {
```

Comment:
super nit: For uniformity `Sr25519Pair` could be used everywhere (or `sr25519::Pair` everywhere), feel free to ignore)

Reply:
**you are right**. Indeed these are the type of things that disturbs be 😆 I change it immediately